### PR TITLE
Add Suffix and LCP arrays algorithms

### PR DESCRIPTION
**Pull request by proserve**
The implementation is correct, but some optimizations should be done (ex: we can replace suffix array by suffix trees to reduce complexity to O(n)),  I will do it in next commits.

### CODE REVIEW

**Comment by proserve** [comment]
Can anyone help me and tell me why my commits have failed in travis check.

**Comment by nryoung** [comment]
See https://github.com/nryoung/algorithms/pull/134#issuecomment-181136918 , the fix is in master already. If you could rebase with master it should fix the Travis failures.

**Comment by proserve** [comment]
Thank you for the comment, i fixed the issue.

File: algorithms/data_structures/lcp_array.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    indices, data compression algorithms and  within the field
+    of bioinformatics.
+
+    for more info : http://algs4.cs.princeton.edu/63suffix/
+    Complexity :
+        worst case : O(n log(n))
+"""
+
+
+def suffix_array(t_string):
```
Comment (nryoung) [inline algorithms/data_structures/lcp_array.py]: Can you add doc strings? Take a look at few of the other modules to get an idea of what should be expected.

File: algorithms/data_structures/lcp_array.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    I use Kasai's algorithm in implementation :
+    Pseudo Code: http://algs4.cs.princeton.edu/32bst
+
+    Complexity :
+     worst case :O(n)
+
+"""
+
+
+def lcp_array(t_str, s_array, rank):
```
Comment (nryoung) [inline algorithms/data_structures/lcp_array.py]: Can you add doc strings? Take a look at few of the other modules to get an idea of what should be expected.

**Comment by nryoung** [comment]
Thanks for your contribution!

Can you also squash and rebase your changes in to a single commit with a short but descriptive commit message of what you are adding? Take a look at https://github.com/edx/edx-platform/wiki/How-to-Rebase-a-Pull-Request if you have any questions on how to do this.

**Comment by nryoung** [comment]
Another thought, can you also add the algorithms to https://github.com/nryoung/algorithms/blob/master/docs/sorting.rst ? This help keep the docs up to date. Please take a look at how the other modules are included if you have questions.

**Comment by proserve** [comment]
Thanks for  your response, i will do the needed work as soon as i have free time

**Comment by proserve** [comment]
done

**Comment by nryoung** [comment]
This looks great! Thanks for your addition.

**Pull request merged**